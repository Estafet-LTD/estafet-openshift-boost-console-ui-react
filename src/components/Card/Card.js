import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

export default styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: white;
  border-radius: 3px;
  padding: 0.5rem;
  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14),
    0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  transition: background 200ms linear;

  ${ifProp(
    'isActive',
    css`
      background: #e5f4f7;
    `
  )};
`;
