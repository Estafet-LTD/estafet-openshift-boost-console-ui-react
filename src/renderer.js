// TODO: Clean up and split up all the different logic handling styled-components, variables set on window object, etc
import { ChunkExtractor, ChunkExtractorManager } from '@loadable/server';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import App from './App';
import configureStore from './store';

const store = configureStore();
const sheet = new ServerStyleSheet();

const statsFile = path.resolve('build/loadable-stats.json');
const extractor = new ChunkExtractor({
  statsFile,
  entrypoints: ['client'],
});

export default (req, res) => {
  const context = {};
  const html = renderToString(
    <ChunkExtractorManager extractor={extractor}>
      <StaticRouter context={context} location={req.url}>
        <Provider store={store}>
          <StyleSheetManager sheet={sheet.instance}>
            <App />
          </StyleSheetManager>
        </Provider>
      </StaticRouter>
    </ChunkExtractorManager>
  );

  const styleTags = sheet.getStyleTags();
  const scriptTags = extractor.getScriptTags();

  if (context.url) {
    res.redirect(context.url);
  } else {
    res.status(200).send(
      `<!doctype html>
      <html lang="">
        <head>
          <meta charset="utf-8" />
          <title>${process.env.PRODUCT_DESCRIPTION}</title>
          <meta name="description" content="" />
          <meta name="theme-color" content="#2BA2B8" />
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <link href="https://fonts.googleapis.com/css?family=Open+Sans&display=swap" rel="stylesheet">
          ${styleTags}
        </head>
        <body>
          <div id="root">${html}</div>
          ${scriptTags}
          <script>
            window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())};
            window.boost = ${JSON.stringify({
              BUILD_API_SERVICE_URI: process.env.BUILD_API_SERVICE_URI,
              TEST_API_SERVICE_URI: process.env.TEST_API_SERVICE_URI,
              PROD_API_SERVICE_URI: process.env.PROD_API_SERVICE_URI,
              PRODUCT_DESCRIPTION: process.env.PRODUCT_DESCRIPTION,
            })};
          </script>
        </body>
      </html>`
    );
  }
};
