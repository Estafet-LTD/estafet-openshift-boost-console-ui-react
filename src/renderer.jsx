import { ChunkExtractor, ChunkExtractorManager } from '@loadable/server';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import App from './App';
import configureStore from './store/configureStore';

const store = configureStore();
const sheet = new ServerStyleSheet();

// Mocking the loadable stats file for the tests
const isTesting = process.env.NODE_ENV === 'test';
const statsFileRoot = isTesting ? 'src/__mocks__' : 'build';
const statsFile = path.resolve(`${statsFileRoot}/loadable-stats.json`);

const extractor = new ChunkExtractor({
  statsFile,
  entrypoints: ['client'],
});

export default (req, res) => {
  const context = {};
  const html = renderToString(
    <ChunkExtractorManager extractor={extractor}>
      <StaticRouter context={context} location={req.url}>
        <Provider store={store}>
          <StyleSheetManager sheet={sheet.instance}>
            <App />
          </StyleSheetManager>
        </Provider>
      </StaticRouter>
    </ChunkExtractorManager>
  );

  const styledComponentsTags = sheet.getStyleTags();
  const styleTags = extractor.getStyleTags();
  const scriptTags = extractor.getScriptTags();

  if (context.url) {
    res.redirect(context.url);
  } else {
    res.status(200).send(
      `<!doctype html>
      <html lang="en">
        <head>
          <meta charset="utf-8" />
          <title>${process.env.PRODUCT_DESCRIPTION}</title>
          <meta name="description" content="" />
          <meta name="theme-color" content="#2BA2B8" />
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" rel="stylesheet">
          ${styleTags}
          ${styledComponentsTags}
        </head>
        <body>
          <div id="root">${html}</div>
          ${scriptTags}
          <script>
            window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())};
            window.__GLOBALS__= {
              PRODUCT: ${JSON.stringify(process.env.PRODUCT)},
              PRODUCT_DESCRIPTION: ${JSON.stringify(process.env.PRODUCT_DESCRIPTION)},
              TASK_MANAGEMENT_TITLE: ${JSON.stringify(process.env.TASK_MANAGEMENT_TITLE)},
              GATEWAY_API_SERVICE_URI: ${JSON.stringify(process.env.GATEWAY_API_SERVICE_URI)},
            };
          </script>
        </body>
      </html>`
    );
  }
};
