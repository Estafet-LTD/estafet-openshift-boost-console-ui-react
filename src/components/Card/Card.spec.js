import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('Components: Card', () => {
  it('it should match the snapshot', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should match the snapshot when active', () => {
    const tree = renderer.create(<Card isActive />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
