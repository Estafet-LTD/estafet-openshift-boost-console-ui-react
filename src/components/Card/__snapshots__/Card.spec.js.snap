// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Card it should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: white;
  border-radius: 3px;
  padding: 0.5rem;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  -webkit-transition: background 200ms linear;
  transition: background 200ms linear;
}

<div
  className="c0"
/>
`;

exports[`Components: Card it should match the snapshot when active 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: white;
  border-radius: 3px;
  padding: 0.5rem;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  -webkit-transition: background 200ms linear;
  transition: background 200ms linear;
  background: #e5f4f7;
}

<div
  className="c0"
/>
`;
